dask/array/core.py

```diff
@@ -3228,54 +3230,104 @@ def auto_chunks(chunks, shape, limit, dtype, previous_chunks=None):
+                    continue
+                elif this_multiplier <= 1:
+                    if max(previous_chunks[a]) == 1:
+                        multiplier_remaining = _trivial_aggregate(a)
+                        chunks[a] = tuple(previous_chunks[a])
+                        continue
+
+                    cache, dimension_result = {}, []
+                    for c in previous_chunks[a]:
+                        if c in cache:
+                            # We potentially split the same chunk over and over again
+                            # so cache that
+                            dimension_result.extend(cache[c])
+                            continue
+                        elif c <= max(max_chunk_size, 1):
+                            new_chunks = [c]
+                        else:
+                            new_chunks = _split_up_single_chunk(
+                                c, this_chunksize_tolerance, max_chunk_size, proposed
+                            )
+
+                        cache[c] = new_chunks
+                        dimension_result.extend(new_chunks)
+
+                        if max(dimension_result) == 1:
+                            # See if we can split up other axes further
+                            multiplier_remaining = _trivial_aggregate(a)
+
                 else:
-                    result[a] = round_to(proposed, ideal_shape[a])
+                    dimension_result, new_chunk = [], 0
+                    for c in previous_chunks[a]:
+                        if c + new_chunk <= proposed:
+                            # keep increasing the chunk
+                            new_chunk += c
+                        else:
+                            # We reach the boundary so start a new chunk
+                            dimension_result.append(new_chunk)
+                            new_chunk = c
+                            if new_chunk > max_chunk_size:
+                                # our new chunk is greater than the allowed, so split it up
+                                dimension_result.extend(
+                                    _split_up_single_chunk(
+                                        c,
+                                        this_chunksize_tolerance,
+                                        max_chunk_size,
+                                        proposed,
+                                    )
+                                )
+                                new_chunk = 0
+                    if new_chunk > 0:
+                        dimension_result.append(new_chunk)
+
+                result[a] = tuple(dimension_result)
 
             # recompute how much multiplier we have left, repeat
-            multiplier = _compute_multiplier(limit, dtype, largest_block, result)
+            if multiplier_remaining:
+                multiplier = _compute_multiplier(
+                    limit, dtype, largest_block, median_chunks
+                )
 
```

Comment:
nit: I've been thinking about the directional `left` here
```suggestion
            if multiplier_remaining:
```

Reply:
done that